orttraining/orttraining/core/agent/training_agent.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <thread>
+#include <future>
+
+#include "core/common/common.h"
+#include "core/common/logging/logging.h"
+#include "core/framework/framework_common.h"
+#include "core/session/IOBinding.h"
+#include "core/session/inference_session.h"
+#include "orttraining/training_ops/cpu/controlflow/ort_tasks.h"
+
+
+namespace onnxruntime {
+namespace training {
+class IOBinding;
+
+class TrainingAgent {
+
+  public:
+    explicit TrainingAgent(InferenceSession* session);
+    virtual ~TrainingAgent();
+    // For ORTModule.forward()
+    virtual common::Status RunForward(const RunOptions& run_options, onnxruntime::IOBinding& io_binding,
+                                               std::vector<OrtValue>& user_outputs,
+                                               int64_t& run_id) ORT_MUST_USE_RESULT;
+    // For ORTModule.backward()
+    common::Status RunBackward(int64_t run_id, const std::vector<OrtValue>& backward_output_grads) ORT_MUST_USE_RESULT;
+    void CancelPendingBackwardRun(int64_t run_id);
+
+  private:
+    // mutex for accessing bg_threads_
+    std::mutex bg_threads_mutex_;
+    // background threads for RunInBackgroundAndWaitForYield and ContinueRunInBackground
```

Comment:
>RunInBackgroundAndWaitForYield and ContinueRunInBackground [](start = 30, length = 58)

these don't appear to be function names, maybe the comment could have more detail?

Reply:
updated.